### PR TITLE
Adding draft of Github flow tutorial

### PR DESCRIPTION
**Pull request by w1stler**
First draft of Github tutorial which might be useful for new members.

### CODE REVIEW

File: github_flow.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        fetch = +refs/heads/*:refs/remotes/origin/*
+[branch "master"]
+        remote = origin
+        merge = refs/heads/master
+[remote "upstream"]
+        url = git@github.com:CodeForPoznan/project_name.git
+        fetch = +refs/heads/*:refs/remotes/upstream/*
+```
+
+Example commands:
```
Comment (magul) [inline github_flow.md]: It should be named: `How to be up-to-date with upstream repository`

File: github_flow.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+1. `git fetch upstream`
+  * To fetch upstream repository (newest changes from master branch from upstream repository - in that case Code for Poznan's main branch for particular project)
+2. `git merge upstream/master`
+  * To merge all the changes from upstream repository to master branch
+3. `git push`
+  * to push all of the changes to forked repository
+
+Flow for own repositories is simple:
```
Comment (magul) [inline github_flow.md]: It should be named `How to propose change in upstream repository`

File: github_flow.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+1. create your own branch 
+  * `git checkout -b name_of_your_branch`
+2. make changes in the code
+3. add the changes
+  * `git add name_of_the_file`
+4. commit the files 
+  * `git commit -m "message that explains the commit"`
+5. push the changes to your own repository
+  * `git push origin name_of_your_branch`
+6. go to Github website and create Pull Request from head eg. w1stler/sample_branch to base codeforpoznan/master
```
Comment (magul) [inline github_flow.md]: `:` instead of `/`

File: github_flow.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+## We have agreed, that we will work using Github flow. Simple instructions can be found below.
+
+In short words, we are forking repository from Code for Poznan organisation to our repository, push the changes there and create pull requests from our branches (eg. w1stler/sample_branch) to codeforpoznan/master. 
+
+## Link for reference: https://guides.github.com/introduction/flow/
+
+Example config for proper Github flow.
+
+```
+[core]
```
Comment (magul) [inline github_flow.md]: I have never (or at least don't remember) changed this file (and You have not even give it name or location).

I would suggest section named `How to setup Your local repository` which will contains:
1. Fork from Code for Poznan
1. Clone Your (forked) repository to Your local computer.
1. Add upstream remote (simply execute command `git remote add upstream it@github.com:CodeForPoznan/project_name.git`

And that should be all

File: github_flow.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+1. create your own branch 
+  * `git checkout -b name_of_your_branch`
+2. make changes in the code
+3. add the changes
+  * `git add name_of_the_file`
+4. commit the files 
+  * `git commit -m "message that explains the commit"`
+5. push the changes to your own repository
+  * `git push origin name_of_your_branch`
+6. go to Github website and create Pull Request from head eg. w1stler:sample_branch to base codeforpoznan:master
```
Comment (magul) [inline github_flow.md]: `` for branch names

File: github_flow.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## Link for reference: https://guides.github.com/introduction/flow/
+
+How to setup your local repository?
+
+1. Fork repository from Code for Poznań
+2. Clone your (forked) repository to local computer
+3. Add upstream remote by executing command ``git remote add upstream git@github.com:CodeForPoznan/project_name.git``
+
+Example config for proper Github flow after changes.
```
Comment (magul) [inline github_flow.md]: so we decided to still have this example here?

File: github_flow.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        fetch = +refs/heads/*:refs/remotes/origin/*
+[branch "master"]
+        remote = origin
+        merge = refs/heads/master
+[remote "upstream"]
+        url = git@github.com:CodeForPoznan/project_name.git
+        fetch = +refs/heads/*:refs/remotes/upstream/*
+```
+
+How to be up-to-date with upstream repository?
```
Comment (magul) [inline github_flow.md]: shouldn't it stat with `#` as section header?

File: github_flow.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+In short words, we are forking repository from Code for Poznan organisation to our repository, push the changes there and create pull requests from our branches (eg. ``w1stler:sample_branch``) to ``codeforpoznan:master``. 
+
+## Link for reference: https://guides.github.com/introduction/flow/
+
+How to setup your local repository?
+
+1. Fork repository from Code for Poznań
+2. Clone your (forked) repository to local computer
+3. Add upstream remote by executing command ``git remote add upstream git@github.com:CodeForPoznan/project_name.git``
```
Comment (magul) [inline github_flow.md]: sometimes You use one ` and sometimes double ` for oneliners. Can we decided and stick to one of them?

File: github_flow.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
+# Github flow
+
+## We have agreed, that we will work using Github flow. Simple instructions can be found below.
+
+In short words, we are forking repository from Code for Poznan organisation to our repository, push the changes there and create pull requests from our branches (eg. ``w1stler:sample_branch``) to ``codeforpoznan:master``. 
+
+## Link for reference: https://guides.github.com/introduction/flow/
+
+How to setup your local repository?
```
Comment (magul) [inline github_flow.md]: seciotn header

File: github_flow.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+## Link for reference: https://guides.github.com/introduction/flow/
+
+### How to setup your local repository?
+
+1. Fork repository from Code for Poznań
+2. Clone your (forked) repository to local computer
+3. Add upstream remote by executing command ``git remote add upstream git@github.com:CodeForPoznan/project_name.git``
+
+### Example config for proper Github flow after changes.
```
Comment (magul) [inline github_flow.md]: do we still want to have this example here?

File: github_flow.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
+# Github flow
+
+## We have agreed, that we will work using Github flow. Simple instructions can be found below.
+
+In short words, we are forking repository from Code for Poznan organisation to our repository, push the changes there and create pull requests from our branches (eg. ``w1stler:sample_branch``) to ``codeforpoznan:master``.
```
Comment (magul) [inline github_flow.md]: you're sometimes using one ` and sometimes double ` for oneliners - can we choose one and stick to it?

**Comment by w1stler** [comment]
@magul, comments addressed. Good catches, thanks! ;-)

File: github_flow.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+In short words, we are forking repository from Code for Poznan organisation to our repository, push the changes there and create pull requests from our branches (eg. `w1stler:sample_branch`) to `codeforpoznan:master`. 
+
+## Link for reference: https://guides.github.com/introduction/flow/
+
+### How to setup your local repository?
+
+1. Fork repository from Code for Poznań
+2. Clone your (forked) repository to local computer
+3. Add upstream remote by executing command `git remote add upstream git@github.com:CodeForPoznan/project_name.git`
```
Comment (Piegusek07) [inline github_flow.md]: please, change remote url to http (instead of ssh)

File: github_flow.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+### How to be up-to-date with upstream repository?
+
+1. `git fetch upstream`
+  * To fetch upstream repository (newest changes from master branch from upstream repository - in that case Code for Poznan's main branch for particular project)
+2. `git merge upstream/master`
+  * To merge all the changes from upstream repository to master branch
```
Comment (Szymanpaulina) [inline github_flow.md]: Missing detail to me is that it is used for my own master

**Pull request merged**